kore/src/Kore/Step/SMT/Translate.hs

```diff
@@ -230,29 +228,39 @@ translatePredicateWith translateTerm predicate =
                 -- will fail to translate.
                 SMT.not <$> translateBool notChild
             ApplySymbolF app ->
-                (<|>)
-                    (translateApplication app)
-                    (translateUninterpreted SMT.tBool pat)
+                translateApplication (Just SMT.tBool) pat app
             DefinedF (Defined child) -> translateBool child
             _ -> empty
 
-    translateApplication :: Application Symbol p -> Translator m variable SExpr
+    translateApplication :: Maybe SExpr -> p -> Application Symbol p -> Translator m variable SExpr
     translateApplication
+        maybeSort
+        original
         Application
             { applicationSymbolOrAlias
             , applicationChildren
             }
-      = do
-        let translated = translateSymbol applicationSymbolOrAlias
-        sexpr <- maybe empty return translated
-        when (isNothing translated)
-            $ warnSymbolSMTRepresentation applicationSymbolOrAlias
-        children <- zipWithM translatePattern
-            applicationChildrenSorts
-            applicationChildren
-        return $ shortenSExpr (applySExpr sexpr children)
+      | isFunctionalPattern original =
+        translateInterpretedApplication
+        <|> translateUninterpreted'
+      | otherwise =
+        translateInterpretedApplication
```

Comment:
Before, we would fall back to `translateUninterpreted` for `translateInt` and `translateBool` without first checking if the pattern is functional. I assumed this wasn't correct, and fixed it here. This is reflected in the tests by the need to add some test data with the `functional` attribute.